### ISSUE TITLE
Fails for not accessing a directory in the parent directory

### ISSUE THREAD

**Issue by Chris00** [ocaml/dune#1060]
I have a directory with a `dune-project` and a `dune` file in a sub-directory `src/`.  When I try to do `dune build`, I get:
```
Error: exception Sys_error("libgksu-Rmtokv: Permission denied")
Backtrace:
Raised by primitive operation at file "src/stdune/path.ml", line 807, characters 25-50
Called from file "src/file_tree.ml", line 181, characters 8-34
Called from file "camlinternalLazy.ml", line 27, characters 17-27
Re-raised at file "camlinternalLazy.ml", line 34, characters 4-11
Called from file "src/file_tree.ml" (inlined), line 122, characters 20-32
Called from file "src/jbuild_load.ml", line 251, characters 16-41
Called from file "src/file_tree.ml", line 139, characters 16-23
Called from file "map.ml", line 315, characters 19-42
Called from file "map.ml", line 315, characters 26-41
Called from file "src/file_tree.ml" (inlined), line 284, characters 2-49
Called from file "src/jbuild_load.ml", line 249, characters 4-252
Called from file "src/main.ml", line 43, characters 4-70
Called from file "src/main.ml", line 275, characters 12-56
Called from file "bin/main.ml", line 794, characters 7-29
Called from file "vendor/cmdliner/src/cmdliner_term.ml", line 27, characters 19-24
Called from file "vendor/cmdliner/src/cmdliner.ml", line 106, characters 32-39
Called from file "vendor/cmdliner/src/cmdliner.ml", line 136, characters 18-36
Called from file "vendor/cmdliner/src/cmdliner.ml", line 251, characters 22-48
Called from file "bin/main.ml", line 1647, characters 10-51
```
The culprit `libgksu-Rmtokv` is a directory in the _parent_ directory that is not accessible (because of permissions) to the user trying to compile. Maybe dune should print a warning but why fail?

**Comment by tomjridge** [comment]
I am also getting this error, but with a segfault! The relevant files are:

```
<git_root>/ocaml/{tjr_blog.opam,dune-project}
<git_root>/ocaml/src/dune
```

So I expect the ocaml/ directory to be identified as the project root. But somehow dune is escaping and examining my filesystem (and then finding a file without read permission).

```
opam exec dune build @install
Error: exception Sys_error("l/2018_q4/new_programming_modules_USE_GITHUB_REPO_INSTEAD/le_new_programming_modules/CO-18-19-prep/JUNK: Permission denied")
Backtrace:
Raised by primitive operation at file "src/stdune/path.ml", line 804, characters 25-50
Called from file "src/file_tree.ml", line 183, characters 8-34
Called from file "camlinternalLazy.ml", line 27, characters 17-27
Re-raised at file "camlinternalLazy.ml", line 34, characters 4-11
Called from file "src/file_tree.ml" (inlined), line 124, characters 20-32
Called from file "src/jbuild_load.ml", line 244, characters 16-41
Called from file "src/file_tree.ml", line 141, characters 16-23
Called from file "map.ml", line 315, characters 19-42
Called from file "map.ml", line 315, characters 19-42
Called from file "map.ml", line 315, characters 26-41
Called from file "map.ml", line 315, characters 26-41
Called from file "map.ml", line 315, characters 19-42
Called from file "map.ml", line 315, characters 19-42
Called from file "map.ml", line 315, characters 26-41
Called from file "map.ml", line 315, characters 19-42
Called from file "map.ml", line 315, characters 26-41
Called from file "map.ml", line 315, characters 26-41
Called from file "map.ml", line 315, characters 26-41
Called from file "map.ml", line 315, characters 19-42
Called from file "map.ml", line 315, characters 26-41
Called from file "map.ml", line 315, characters 26-41
Called from file "src/file_tree.ml" (inlined), line 286, characters 2-49
Called from file "src/jbuild_load.ml", line 242, characters 4-282
Called from file "src/main.ml", line 44, characters 4-70
Called from file "src/main.ml", line 281, characters 12-56
Called from file "bin/main.ml", line 131, characters 4-26
Called from file "bin/main.ml", line 146, characters 33-42
Called from file "vendor/cmdliner/src/cmdliner_term.ml", line 27, characters 19-24
Called from file "vendor/cmdliner/src/cmdliner.ml", line 106, characters 32-39
Called from file "vendor/cmdliner/src/cmdliner.ml", line 136, characters 18-36
Called from file "vendor/cmdliner/src/cmdliner.ml", line 251, characters 22-48
Called from file "bin/main.ml", line 1092, characters 10-51

I must not segfault.  Uncertainty is the mind-killer.  Exceptions are
the little-death that brings total obliteration.  I will fully express
my cases.  Execution will pass over me and through me.  And when it
has gone past, I will unwind the stack along its path.  Where the
cases are handled there will be nothing.  Only I will remain.
make: *** [Makefile:4: build] Error 1

```

**Comment by tomjridge** [comment]
Actually I just realized there was a dune_project in my /tmp directory, hence crazy behaviour. Please ignore!

**Comment by diml** [comment]
Well, there is definitely an issue here as Dune shouldn't crash with a backtrace.

**Comment by tomjridge** [comment]
Yes, sure. Strange indeed that I also get a segfault.

**Comment by liuchong** [comment]
similar error here:
```
dune build hello_world.exe
Error: exception Sys_error("pulse-PKdhtXMmr18n: Permission denied")
Backtrace:
Raised by primitive operation at file ... ... ...
```

**Comment by rgrinberg** [comment]
Could you please test PR #2004 to see if it addresses these issues?

**Comment by rgrinberg** [comment]
@tomjridge which version of dune did you use to observe that exception?

**Comment by tomjridge** [comment]
@rgrinberg Sorry - I should have noted this in the original report. The computer I reported that from is no longer available, so I can't be sure what the dune version was. 

Also, now I actually read the error message, the "I must not segfault... " is not a segfault. So apologies for the very poor report.

**Comment by rgrinberg** [comment]
Fixed in 1.9
